### PR TITLE
Remove arguments/`sys.argv` from pre/post_command actions

### PR DESCRIPTION
**Pull request by kenodegard**
### Description

Since we haven't solved the [inconsistency](https://github.com/conda/conda/pull/12814) of builtin subcommands passing a parsed `Namespace` and plugin subcommands passing an iterable of strings (i.e., `sys.argv`) this removes passing in the args into the pre/post command until we can properly identify a safe path forward.

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [x] Add / update outdated documentation?

### CODE REVIEW

File: conda/plugins/manager.py [hunk 1]
```diff
@@ -167,19 +166,25 @@ def get_solver_backend(self, name: str = None) -> type[Solver]:
 
         return backend
 
-    def yield_command_hook_actions(self, hook_type: CommandHookTypes, command: str):
+    def apply_pre_commands(self, command: str) -> None:
```
Comment (jezdez) [inline conda/plugins/manager.py:169]:
```suggestion
    def invoke_pre_commands(self, command: str) -> None:
```

If the docstring uses that term, let's use that verb :D

File: conda/cli/conda_argparse.py [hunk 2]
```diff
@@ -117,9 +116,9 @@ def do_call(arguments: argparse.Namespace, parser: ArgumentParser):
     """
     # First, check if this is a plugin subcommand; if this attribute is present then it is
     if getattr(arguments, "plugin_subcommand", None):
-        _run_command_hooks("pre", arguments.plugin_subcommand.name, sys.argv[2:])
+        context.plugin_manager.apply_pre_commands(arguments.plugin_subcommand.name)
```
Comment (jezdez) [inline conda/cli/conda_argparse.py:119]:
```suggestion
        context.plugin_manager.invoke_pre_commands(arguments.plugin_subcommand.name)
```

File: conda/cli/conda_argparse.py [hunk 3]
```diff
@@ -117,9 +116,9 @@ def do_call(arguments: argparse.Namespace, parser: ArgumentParser):
...
         result = arguments.plugin_subcommand.action(sys.argv[2:])
-        _run_command_hooks("post", arguments.plugin_subcommand.name, sys.argv[2:])
+        context.plugin_manager.apply_post_commands(arguments.plugin_subcommand.name)
```
Comment (jezdez) [inline conda/cli/conda_argparse.py:121]:
```suggestion
        context.plugin_manager.invoke_post_commands(arguments.plugin_subcommand.name)
```

File: conda/cli/conda_argparse.py [hunk 4]
```diff
@@ -130,27 +129,13 @@ def do_call(arguments: argparse.Namespace, parser: ArgumentParser):
     module = import_module(relative_mod, __name__.rsplit(".", 1)[0])
 
     command = relative_mod.replace(".main_", "")
-    _run_command_hooks("pre", command, arguments)
+    context.plugin_manager.apply_pre_commands(command)
```
Comment (jezdez) [inline conda/cli/conda_argparse.py:132]:
```suggestion
    context.plugin_manager.invoke_pre_commands(command)
```

File: conda/cli/conda_argparse.py [hunk 5]
```diff
@@ -130,27 +129,13 @@ def do_call(arguments: argparse.Namespace, parser: ArgumentParser):
...
     result = getattr(module, func_name)(arguments, parser)
-    _run_command_hooks("post", command, arguments)
+    context.plugin_manager.apply_post_commands(command)
```
Comment (jezdez) [inline conda/cli/conda_argparse.py:134]:
```suggestion
    context.plugin_manager.invoke_post_commands(command)
```

**Review by jezdez — APPROVED** [review]
Just minor nitpicking

File: tests/plugins/test_pre_command.py [hunk 6]
```diff
@@ -57,7 +53,7 @@ def test_pre_command_action_raises_exception(pre_command_plugin, conda_cli):
     that it bubbles up to the top and isn't caught anywhere. This will ensure that it
     goes through our normal exception catching/reporting mechanism.
     """
-    exc_message = "Boom!"
+    exc_message = "💥"
```
[the remark this answers is not in the document]
Comment (kenodegard) [reply tests/plugins/test_pre_command.py:56]: hehe just to make the pre and post tests symmetrical

File: conda_env/cli/main.py [hunk 7]
```diff
@@ -49,10 +49,10 @@ def do_call(arguments, parser):
     # Run the pre_command actions
     command = relative_mod.replace(".main_", "")
 
-    _run_command_hooks("pre", f"env_{command}", arguments)
+    context.plugin_manager.apply_pre_commands(f"env_{command}")
     module = import_module(relative_mod, __name__.rsplit(".", 1)[0])
     exit_code = getattr(module, func_name)(arguments, parser)
-    _run_command_hooks("post", f"env_{command}", arguments)
+    context.plugin_manager.apply_post_commands(f"env_{command}")
```
Comment (kenodegard) [inline conda_env/cli/main.py:55]:
```suggestion
    context.plugin_manager.invoke_post_commands(f"env_{command}")
```

**Review by jezdez — DISMISSED** [review]

**Review by travishathaway — CHANGES_REQUESTED** [review]
@kenodegard, the documentation still needs to be updated:

- https://docs.conda.io/projects/conda/en/latest/dev-guide/plugins/post_commands.html
- https://docs.conda.io/projects/conda/en/latest/dev-guide/plugins/pre_commands.html

Those two pages still talk about passing the pre/post hook functions arguments.

Other than that, this pull request looks good to me.

**Review by beeankha — APPROVED** [review]

**Review by travishathaway — APPROVED** [review]
LGTM 🚀

**Pull request merged**